Implement LEQs `interpreter\src\check.rs`

```
PS C:\Users\lucas\dev\idm> cargo build -p interpreter   
   Compiling interpreter v0.1.0 (C:\Users\lucas\dev\idm\interpreter)
error[E0004]: non-exhaustive patterns: `&ast::CondExpr::Leq(_, _)` not covered
   --> interpreter\src\check.rs:103:11
    |
103 |     match cond_expr {
    |           ^^^^^^^^^ pattern `&ast::CondExpr::Leq(_, _)` not covered
    |
note: `ast::CondExpr` defined here
   --> C:\Users\lucas\dev\idm\ast\src\lib.rs:40:1
    |
40  | pub enum CondExpr {
    | ^^^^^^^^^^^^^^^^^
...
44  |     Leq(Expr, Expr),
    |     --- not covered
    = note: the matched value is of type `&ast::CondExpr`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
106 ~         CondExpr::Eq(e1, e2) => check_expression(e1, globals, locals).and(check_expression(e2, globals, locals)),
107 ~         &ast::CondExpr::Leq(_, _) => todo!(),
    |

error[E0004]: non-exhaustive patterns: `&ast::LogicExpr::Leq(_, _)` not covered
   --> interpreter\src\check.rs:115:11
    |
115 |     match logic_expr {
    |           ^^^^^^^^^^ pattern `&ast::LogicExpr::Leq(_, _)` not covered
    |
note: `ast::LogicExpr` defined here
   --> C:\Users\lucas\dev\idm\ast\src\lib.rs:56:1
    |
56  | pub enum LogicExpr {
    | ^^^^^^^^^^^^^^^^^^
...
60  |     Leq(LogicInt, LogicInt),
    |     --- not covered
    = note: the matched value is of type `&ast::LogicExpr`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
118 ~         LogicExpr::Eq(e1, e2) => check_logic_int(e1, locals).and(check_logic_int(e2, locals)),
119 ~         &ast::LogicExpr::Leq(_, _) => todo!(),
    |

For more information about this error, try `rustc --explain E0004`.
error: could not compile `interpreter` (lib) due to 2 previous errors
```

--- a/interpreter/src/check.rs
+++ b/interpreter/src/check.rs
@@ -104,6 +104,7 @@ fn check_cond_expr(cond_expr: &CondExpr, globals: &HashSet<String>, locals: &Has
         CondExpr::Neg(e) => check_cond_expr(e, globals, locals),
         CondExpr::And(e1, e2) => check_cond_expr(e1, globals, locals).and(check_cond_expr(e2, globals, locals)),
         CondExpr::Eq(e1, e2) => check_expression(e1, globals, locals).and(check_expression(e2, globals, locals)),
+        CondExpr::Leq(e1, e2) => check_expression(e1, globals, locals).and(check_expression(e2, globals, locals)),  
     }
 }
 
@@ -116,6 +117,7 @@ fn check_logic_expr(logic_expr: &LogicExpr, locals: &HashMap<String, HashSet<Str
         LogicExpr::Neg(e) => check_logic_expr(e, locals),
         LogicExpr::And(e1, e2) => check_logic_expr(e1, locals).and(check_logic_expr(e2, locals)),
         LogicExpr::Eq(e1, e2) => check_logic_int(e1, locals).and(check_logic_int(e2, locals)),
+        LogicExpr::Leq(e1, e2) => check_logic_int(e1, locals).and(check_logic_int(e2, locals)),
     }
 }
 

